clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerNetworkClient.java

```diff
@@ -183,25 +183,28 @@ public void poll(long timeout) {
 
     private void poll(long timeout, long now) {
         // send all the requests we can send now
-        pollUnsentRequests(now);
-        now = time.milliseconds();
-        
+        trySend(now);
+
         // ensure we don't poll any longer than the deadline for
         // the next scheduled task
         timeout = Math.min(timeout, delayedTasks.nextTimeout(now));
         clientPoll(timeout, now);
+        now = time.milliseconds();
+
+        // handle any disconnects by failing the active requests. note that disconects must
+        // be checked immediately following poll since any subsequent call to client.ready()
+        // will reset the disconnect status
+        checkDisconnects(now);
 
         // execute scheduled tasks
-        now = time.milliseconds();
         delayedTasks.poll(now);
 
         // try again to send requests since buffer space may have been
         // cleared or a connect finished in the poll
-        pollUnsentRequests(now);
+        trySend(now);
```

Comment:
Is there any downside to no longer waiting until either `NetworkClient` is ready or the node is disconnected?